google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb

```diff
@@ -372,7 +386,7 @@ def partition_read table, columns, keys: nil, index: nil,
         #
         #   batch_snapshot.close
         #
-        def execute_partition partition, call_options: nil
+        def execute_partition partition, call_options: nil, directed_read_options: nil
```

Comment:
@NivedhaSenthil 
`execute_partition` method takes partition object as input, which already has the request body (ExecuteSqlRequest or ReadRequest created). We need not expose this option separately in execute_partition and execute_partition_query.

**Can you change this to below implementation,**
1. Expose a new field for directed_read_options in [partition_query](https://github.com/googleapis/ruby-spanner/blob/ae6cf4509b11fb6f1e1b414d52c158c51e792ef1/google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb#L215) similar to `data_boost_enabled`.
2. Set the input option to ExecuteSqlRequest body like [here](https://github.com/googleapis/ruby-spanner/blob/ae6cf4509b11fb6f1e1b414d52c158c51e792ef1/google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb#L238)
3. Remove the input changes in [execute_partition](https://github.com/googleapis/ruby-spanner/blob/ae6cf4509b11fb6f1e1b414d52c158c51e792ef1/google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb#L389C13-L389C30) and [execute_partition_query](https://github.com/googleapis/ruby-spanner/blob/ae6cf4509b11fb6f1e1b414d52c158c51e792ef1/google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb#L849) 
4. Set the directedreadoption from partition [here](https://github.com/googleapis/ruby-spanner/blob/ae6cf4509b11fb6f1e1b414d52c158c51e792ef1/google-cloud-spanner/lib/google/cloud/spanner/batch_snapshot.rb#L861)

Reply:
You can take data_boost_enabled sample as reference. The same would aplly for directed_read_options
https://cloud.google.com/spanner/docs/samples/spanner-batch-client#spanner_batch_client-ruby.

Also make this change for partition_read also.

Reply:
Thanks